cmd/download.go

```diff
@@ -11,22 +19,200 @@ var downloadCmd = &cobra.Command{
 	Use:     "download",
 	Aliases: []string{"d"},
 	Short:   "Download an exercise.",
-	Long: `Download an Exercism exercise to work on.
+	Long: `Download an exercise.
 
-If you've already started working on the exercise, the command will
-also download your most recent solution.
+You may download an exercise to work on. If you've already
+started working on it, the command will also download your
+latest solution.
 
-This does not automatically overwrite identically named files.
-If it finds a file with the same name, it will ask if you want to overwrite it.
-Or you can overwrite files automatically by passing the --force flag.
-
-Download other people's solution by providing the UUID.
+Download other people's solutions by providing the UUID.
 `,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("download called")
+		uuid, err := cmd.Flags().GetString("uuid")
+		BailOnError(err)
+		if uuid == "" && len(args) == 0 {
+			// TODO: usage
+			fmt.Fprintf(os.Stderr, "need an exercise name or a solution --uuid")
+			return
+		}
+
+		var exercise string
+		if len(args) > 0 {
+			exercise = args[0]
+		}
+
+		track, err := cmd.Flags().GetString("track")
+		BailOnError(err)
+
+		dr, err := newDownloadRequest(uuid, track, exercise)
+		BailOnError(err)
+
+		res, err := dr.do()
+		BailOnError(err)
+
+		if res.StatusCode != http.StatusOK {
+			switch dr.payload.Error.Type {
+			case "track_ambiguous":
+				// TODO: interactive selection
+				fmt.Fprintf(os.Stderr, "You have multiple %s exercises available to you.\n", solution.Exercise)
+				fmt.Fprintln(os.Stderr, "Specify the the --track flag:")
+				for _, id := range dr.payload.Error.PossibleTrackIDs {
+					fmt.Fprintf(os.Stderr, "%s download %s --track=%s", BinName, solution.Exercise, id)
```

Comment:
s/BinName/BinaryName ?

Reply:
Gah. I must not have run the tests the final time after resolving some conflicts. That's terrible.